File: 0_sliding_window/subarraysWithKDistinct/brute.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var subarraysWithKDistinct = function (nums, k) {
  let cnt = 0;
  for (let i = 0; i < nums.length; i++) {
    let set = new Set();
    for (let j = i; j < nums.length; j++) {
      set.add(nums[j]);
      if (set.size == k) {
        cnt++;
      }
      if (set.size > k) {
        break;
      }
    }
  }
  return cnt;
};
const r = subarraysWithKDistinct([1, 2, 1, 2, 3], 2);
console.log(`  `, r);

File: 0_array/question/longestConsecutive/brute.cpp

#include <iostream>
#include <map>
#include <vector>
using namespace std;
class Solution {
public:
    bool linearSearch(vector<int>& a, int num) {
        int n = a.size(); 
        // Traverse through the array 
        for (int i = 0; i < n; i++) {
            if (a[i] == num)
                return true;
        }
        return false;
    }
    int longestConsecutive(vector<int>& nums) {
           if (nums.size() == 0) {
            return 0;
        }
        int n = nums.size();
        int longest = 1; 
        for (int i = 0; i < n; i++) {
               int x = nums[i]; 
                int cnt = 1; 
           while (linearSearch(nums, x + 1) == true) {
                //!:: unable to think Move to the next number in the sequence
                x += 1; 
                cnt += 1; 
            }
            longest = max(longest, cnt);

        }
        return longest;

    }
};